src/build_workflow/build_incremental.py

```diff
@@ -0,0 +1,64 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+from typing import List
+
+from manifests.build_manifest import BuildManifest
+from manifests.input_manifest import InputManifest
+
+
+class BuildIncremental:
+    def __init__(self, input_manifest: InputManifest, distribution: str):
+        self.distribution = distribution
+        self.input_manifest = input_manifest
+
+    # Given input manifest and return a list of what components changed and added.
+    def commits_diff(self, input_manifest: InputManifest) -> List[str]:
+        build_manifest_path = os.path.join(self.distribution, "builds", input_manifest.build.filename, "manifest.yml")
+        if not os.path.exists(build_manifest_path):
+            logging.info("Previous build manifest does not exist. Rebuilding Core.")
+            return [input_manifest.build.name.replace(" ", "-")]
+        previous_build_manifest = BuildManifest.from_path(build_manifest_path)
+        stable_input_manifest = input_manifest.stable()
+        if previous_build_manifest.build.version != stable_input_manifest.build.version:
```

Comment:
Why this check is here?
Could there be a case where the previous build manifest for an OS version differ from that in input manifest?
If yes, can you please give an example.

Reply:
This came out of my mind when I was testing on local. When we are building artifacts in local, we don't have any sub-folder for different version. All built artifacts stored in the same directory following this pattern: `<architecture>/builds/<opensearch/opensearch-dashboards>/<plugins/... or manifest.yml>`. Even we built input manifests of different versions. Latter build-manifest will overwrite the previous one but all artifacts would all be in place without any of removal. 

So I believe at least we need to check we are incrementally building the same version artifacts.